xml2rfc_compat/tests/test_serializers.py

```diff
@@ -488,6 +554,130 @@ def test_create_author_with_non_editor_role(self):
+                },
+                "affiliation": [
+                    {
+                        "organization": {
+                            "name": {
+                                "content": "Internet Research Task Force",
+                                "language": "en",
+                            },
+                        }
+                    },
+                    {
+                        "organization": {
+                            "name": {
+                                "content": "Internet Engineering Task Force",
+                                "language": "en",
+                            },
+                        }
+                    },
+                ],
+            },
+            "role": [
+                {
+                    "type": "author",
+                }
+            ],
+        }
+        author: etree._Element = create_author(Contributor(**person))
+        self.assertEqual(author.organization, "Internet Research Task Force")
+
+    def test_create_author_org_address(self):
+        org: dict[str, Any] = {
+            "organization": {
+                "name": {
+                    "content": "Internet Engineering Task Force",
+                    "language": "en",
+                },
+                "contact": [
+                    {"address": {"country": "United States", "city": "Santa Cruz"}},
+                    {"address": {"country": "United States"}},
+                ],
+                "url": "www.ietf.org",
+            },
+            "role": [
+                {
+                    "type": "publisher",
+                }
+            ],
+        }
+        author: etree._Element = create_author(Contributor(**org))
+        self.assertTrue(hasattr(author, "address"))
+        self.assertTrue(hasattr(author.address, "postal"))
+        self.assertTrue(hasattr(author.address.postal, "country"))
+        self.assertEqual(author.address.postal.country, "United States")
+        self.assertTrue(hasattr(author.address.postal, "city"))
+        self.assertEqual(author.address.postal.city, "Santa Cruz")
+        self.assertTrue(hasattr(author.address, "uri"))
+        self.assertEqual(author.address.uri, "www.ietf.org")
+
+    def test_create_author_missing_complete_name(self):
+        person: dict[str, Any] = {
+            "person": {
+                "name": {
+                    "prefix": {"content": "Dr", "language": "en"},
+                    "given": {
+                        "formatted_initials": {"content": "V G", "language": "en"}
+                    },
+                    "surname": {"content": "Cerf", "language": "en"},
+                },
+            },
+            "role": [
+                {
+                    "type": "author",
+                }
+            ],
+        }
+        author: etree._Element = create_author(Contributor(**person))
+        self.assertEqual(author.get("fullname"), "Dr V GCerf")
```

Comment:
Note that there's no space between the initials, `V G`, and the surname, `Cerf`. i.e. 

- `Dr V GCerf` instead of 
- `Dr V G Cerf`

https://github.com/microamp/bibxml-service/blob/main/xml2rfc_compat/serializers/authors.py#L159-L165

I'm not sure there should've been?

This only happens when `completename` is omitted and `formatted_initials` is provided.

Reply:
In `SerializerTestCase.setUp`, there is
```python
self.contributor_person_data: Dict[str, Any] = {
    "person": {
        "name": {
            "given": {
                "formatted_initials": {
                    "content": "Mr",
                    "language": "en",
                },
            },
            "surname": {"content": "Cerf", "language": "en"},
            "completename": {"content": "Mr Cerf", "language": "en"},
        },
    },
    "role": [{
        "type": "author",
    }],
}
self.contributor_person = Contributor(**self.contributor_person_data)
```
which implies that there should be a space between `formatted_initials` and `surname`. I'm not sure if this is a good example though.

Possibly a bug, I'll create a separate issue for further discussions.